### ISSUE TITLE
Ability to Print or Export a whole notebook

### ISSUE THREAD

**Issue by fmartelg** [notable/notable#600]
### Feature description

As a writer of documentation for a team, I want to be able to print / export all the notes in a notebook as one pdf, html, or markdown document, so that I can share the complete notebook with colleagues.

### Feature motivation

Save time and communicate more effectively.  For example, today how would I print out the whole Notable Tutorial as one document in one go?

**Comment by fabiospampinato** [comment]
You can export a whole notebook by selecting all the notes in it and exporting them.

Exporting to a single page/pdf isn't supported though, is this that you need or is exporting all notes individually at once enough?

**Comment by fmartelg** [comment]
I would like to export everything as one document.  For example, imagine your Notable Tutorial is a manual where each note is a chapter in the manual.  I want to print the manual - the whole manual, including all chapters - as one document, and share it in a training workshop.

**Comment by fabiospampinato** [comment]
Makes sense 👍 There isn't a way to do this currently, but you could merge multiple PDFs together today with some third-party tool for instance (if you are on macOS you can define a service that allows you to combine multiple PDFs with a right click).

**Comment by fmartelg** [comment]
Yes, thank you.  I am on Windows, and there are tools for that.  But it would be nice to have it as one click in Notable.

BTW great work!  I love the simplicity and effectiveness of it all.

**Comment by fabiospampinato** [comment]
I'm glad you like it :) 

I think I'll implement this once a command panel #338 (or at least an API for it) is implemented, it may take a while to get there, but I'd like to make the app extendable #128 and ideally to make as much as possible what's implemented in the core implementable via plugins too, so I'd rather not code some custom, non general, UI for this.

**Comment by RichardGladman** [comment]
I'd like to add my vote for this feature. It would be far more convenient than exporting them and trying to remember ghost script syntax.

**Comment by fabiospampinato** [comment]
@RichardGladman I'd recommend you to add your ":+1:" to the original post, as then I can sort issues by the number of ":+1:"s they got.

**Comment by alpha-beta-soup** [comment]
In the case of HTML export, my preference would be to just capture the `@note/linked-document.md` references, and also export those, so that links between documents are preserved. At present if I use a link like that in a "main" document, and export it *alongside* the linked document, the link still points to `~/notable/notes/document.md` instead of `./document.html`. It would be nice if these links could be adapted as part of the export process.

**Comment by fabiospampinato** [comment]
@alpha-beta-soup that's a bit of a separate issue I think. I'm not sure if we should export all linked notes automatically, but it makes sense that all links should keep working 🤔

**Comment by alpha-beta-soup** [comment]
To be fair, then links do keep working *on my machine*. But the point of export seems to be making the document sharable and self-contained. Maybe it's a separate issue, but IMO a PDF export of a document with links shouldn't attempt to insert content into the main document—I typically use links to move less important matter into addendums and asides—they should be exported as independent PDFs so links don't break. Ditto for HTML export.
